packages/@uppy/react/src/DashboardModal.js

```diff
@@ -37,90 +37,15 @@ class DashboardModal extends Component {
   }
 
   installPlugin () {
-    const {
-      id = 'react:DashboardModal',
-      uppy,
-      target,
-      open,
-      onRequestClose,
-      closeModalOnClickOutside,
-      disablePageScrollWhenModalOpen,
-      inline,
-      plugins, // eslint-disable-line no-shadow
-      width,
-      height,
-      showProgressDetails,
-      note,
-      metaFields, // eslint-disable-line no-shadow
-      proudlyDisplayPoweredByUppy,
-      autoOpenFileEditor,
-      animateOpenClose,
-      browserBackButtonClose,
-      closeAfterFinish,
-      disableStatusBar,
-      disableInformer,
-      disableThumbnailGenerator,
-      disableLocalFiles,
-      disabled,
-      hideCancelButton,
-      hidePauseResumeButton,
-      hideProgressAfterFinish,
-      hideRetryButton,
-      hideUploadButton,
-      showLinkToFileUploadResult,
-      showRemoveButtonAfterComplete,
-      showSelectedFiles,
-      waitForThumbnailsBeforeUpload,
-      fileManagerSelectionType,
-      theme,
-      thumbnailType,
-      thumbnailWidth,
-      locale, // eslint-disable-line no-shadow
-    } = this.props
+    const { id='react:DashboardModal', target=this.container, open, onRequestClose, uppy } = this.props
```

Comment:
Not super fond of eslint's requirement for object destructuring tbh, this code was a much easier read before I needed to rewrite it into `{a, b, c} = this.props`

Reply:
It's quite important, you're making it super easy to pass many needless props causing unnecessary re-renders. Only exception being the top level component, where we do have to always pass all of them in.

Reply:
That being this particular example is probably fine to spread the props indeed.

Reply:
Yea we do the same in the nearby <Dashboard/> component
Both can be refactored to avoid passing extra props + propTypes should be factored out in both components (and even removed? we have types now), but avoided doing it in this pr